jax/api.py

```diff
@@ -1039,10 +1045,12 @@ def vmap(fun: Callable[..., T], in_axes=0, out_axes=0, axis_name=None) -> Callab
       of the mapped input axes for all mapped positional arguments must all
       be equal.
 
-    out_axes: A non-negative integer, None, or (nested) standard Python container
+    out_axes: An integer, None, or (nested) standard Python container
       (tuple/list/dict) thereof indicating where the mapped axis should appear
       in the output. All outputs with a mapped axis must have a non-None
-      ``out_axes`` specification.
+      ``out_axes`` specification. Axis integers must be
+      in the range ``[-ndim, ndim)`` for each output array, where ``ndim`` is
```

Comment:
If -1 means something different from ndim-1, i.e. if -1 means "place so that it's at the end" while ndim-1 means "place so that it's second-to-last", is that worth calling out here? (Or maybe that's not the convention).

Furthermore, does it make sense to allow values in [-ndim, ndim] so that -1 means the same as ndim, or should 'place last' only be expressible with negative axis indices?

Reply:
But `-1` does mean the same as `ndim-1`. But `ndim` is the number of dimensions of the array *returned* by `vmap`, not of the array returned by the vmap-ed function. Let me add some text to make that clearer.

`[-ndim,ndim]` would not make sense; `[-ndim, ndim)` is the legal set of axes. You can also say "last" for a fixed rank with `ndim - 1`, but again note the `ndim` is the `vmap`-output's rank.

Reply:
Done. Is it clear now?

Reply:
Ah yes thanks, that clears up my confusion (still thinking in terms of `np.insert`). Nice!